components/utils.py

```diff
@@ -1222,4 +1223,6 @@ def verify_service_http(service_name, url, predicate=None):
     try:
         return check_http_response(url, predicate)
     except (IOError, ValueError) as e:
+        if e.args[1] == 401:
```

Comment:
is this needed? Please note that the predicate function (eg. the function that checks `.response_code in {401}`), is passed into `check_http_response` as `predicate`. It doesn't expect the response be returned from `verify_service_http` (i think currently nothing uses the return value from `verify_service_http`).
Btw, if we're to use requests, we might want to switch the IOError for `requests.exceptions.ConnectionError`?